src/fenics_constitutive/solver/corotational_solver.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+import dolfinx as df
+from petsc4py import PETSc
+from ._solver import IncrSmallStrainProblem
+from .corotational_lawonsubmesh import CorotationalLawOnSubMesh
+from dolfinx.fem.petsc import NonlinearProblem
+
+class CorotationalIncrSmallStrainProblem(IncrSmallStrainProblem):
+    """
+    Corotational extension of IncrSmallStrainProblem that adds objective
+    stress-rate rotation and mesh-update steps.
+
+    This subclass reuses all core functionality of the base solver and
+    overrides only the initialization and `form` method.
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+
+        super().__init__(*args, **kwargs)
+
+        self._check_isoparametric()
+
+        # Reference configuration
+        self._X_ref = self._u.function_space.mesh.geometry.x.copy()
+
+        # Replace each base LawOnSubMesh with its corotational variant
+        # while preserving all submesh data and function references
+        self._law_on_submeshs = [
+            CorotationalLawOnSubMesh(
+                law=lom.law,
+                cells=lom.cells,
+                displacement_gradient_fn=lom.displacement_gradient_fn,
+                stress=lom.stress,
+                local_tangent=lom.local_tangent,
+                submesh_map=lom.submesh_map,
+                history=lom.history,
+            )
+            for lom in self._law_on_submeshs
+        ]
+
+    @df.common.timed("constitutive-form-evaluation-corotational")
+    def form(self, x: PETSc.Vec) -> None:
+        """
+        Assemble the nonlinear form using a corotational midpoint configuration.
+
+        This override performs a mesh update to the midpoint configuration
+        and applies a corotational stress rotation before evaluating the constitutive
+        laws. The mesh is updated to final configuration prior to scattering stress and tangent fields.
+        """
+        NonlinearProblem.form(self, x)
+
+        self._move_mesh_previous_to_midpoint()
+
+        self.incr_disp.update_current(x)
+
+        for law in self._law_on_submeshs:
+            law.evaluate(self.sim_time, self.incr_disp, self.stress, self.tangent)
+
+        self._move_mesh_midpoint_to_final()
+
+        self.stress.scatter_current()
+        self.tangent.x.scatter_forward()
+
+    def _move_mesh_previous_to_midpoint(self):
+        """Moves mesh geometry to midpoint configuration."""
+
+        # displacement increment needed to move to midpoint
+        self._midpoint_disp = 0.5 * (self._u.x.array - self._u0.x.array)
+
+        # Update the reference configuration to midpoint
+        mesh = self._u.function_space.mesh
+        mesh.geometry.x[:] = self._X_ref + self._u0.x.array.reshape(-1, 3) +  self._midpoint_disp.reshape(-1, 3)
```

Comment:
`midpoint_disp` does not need to be an attribute of the class
```suggestion
        midpoint_disp = 0.5 * (self._u.x.array - self._u0.x.array)

        # Update the reference configuration to midpoint
        mesh = self._u.function_space.mesh
        mesh.geometry.x[:] = self._X_ref + self._u0.x.array.reshape(-1, 3) +  midpoint_disp.reshape(-1, 3)
```